hoomd/hpmc/field.py

```diff
@@ -32,208 +37,142 @@ def __init__(self):
+class Harmonic(ExternalField):
+    r"""Restrain particle positions and orientations with harmonic springs
 
-    * **lattice_energy** -- total lattice energy
-    * **lattice_energy_pp_avg** -- average lattice energy per particle multiplied by the spring constant
-    * **lattice_energy_pp_sigma** -- standard deviation of the lattice energy per particle multiplied by the spring constant
-    * **lattice_translational_spring_constant** -- translational spring constant
-    * **lattice_rotational_spring_constant** -- rotational spring constant
-    * **lattice_num_samples** -- number of samples used to compute the average and standard deviation
+    Args:
+        reference_positions ((*N_particles*, 3) `numpy.ndarray` of
+            `float`): the reference positions, to which particles are restrained
+            :math:`[\mathrm{length}]`.
+        reference_orientations ((*N_particles*, 4) `numpy.ndarray` of
+            `float`): the reference orientations, to which particles are
+            restrained :math:`[\mathrm{dimensionless}]`.
+        k_translational (`float`): translational spring constant
+            :math:`[\mathrm{energy} \cdot \mathrm{length}^{-2}]`.
+        k_rotational (`float`): rotational spring constant
+            :math:`[\mathrm{energy}]`.
+        symmetries ((*N_sym*, 4) `numpy.ndarray` of
+            `float`): the orientations that are equivalent through symmetry,
+            i.e., the rotation quaternions that leave the particles unchanged
+            :math:`[\mathrm{dimensionless}]`.
+
+    :py:class:`Harmonic` specifies that harmonic springs are used to
+    restrain the position and orientation of every particle:
 
-    .. warning::
-        The lattice energies and standard deviations logged by this class are multiplied by the spring constant.
+    .. math::
 
-    Example::
+        V_{i}(r_i)  = \frac{1}{2} k_{trans} \cdot (r_i-r_{0,i})^2 \\
+        V_{i}(q_i)  = \frac{1}{2} k_{rot} \cdot (q_i-q_{0,i})^2
 
-        mc = hpmc.integrate.sphere(seed=415236);
-        hpmc.field.lattice_field(mc=mc, position=fcc_lattice, k=1000.0);
-        log = analyze.log(quantities=['lattice_energy'], period=100, filename='log.dat', overwrite=True);
+    where :math:`k_{trans}` and :math:`k_{rot}` correspond to the arguments
+    ``k_translational`` and ``k_rotational``, respectively, :math:`r_i` and
+    :math:`q_i` are the position and orientation of particle :math:`i`, and the
+    :math:`0` subscripts denote the given reference quantities. When multiple
+    orientations are passed to ``symmetries``, the rotational energy for each
+    particle is computed for not only the orientation of the particle at its
+    current orietation, but also at the orientation resulting from the rotation
+    represented by each quaternion in ``symmetries`` yielding a list of
+    energies :math:`\mathbb{V}`; the contribution to the energy is the minimum
+    value in this list, :math:`\mathrm{min}(\mathbb{V})`.
 
```

Comment:
Need an `Attributes:` section that documents the `param_dict` entries. It can mostly be copied and pasted from `Args:`.